script/utils/CurrentConfigCheck.s.sol

```diff
@@ -0,0 +1,570 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "./ExistingDeploymentParser.sol";
+import "./TimelockEncoding.sol";
+
+/**
+ * forge script script/utils/CurrentConfigCheck.s.sol:CurrentConfigCheck -vvv --sig "run(string)" $NETWORK_NAME
+ * NETWORK_NAME options are currently preprod-holesky, testnet-holesky, mainnet, local
+ */
+contract CurrentConfigCheck is ExistingDeploymentParser, TimelockEncoding {
+    string deployedContractsConfig;
+    string intialDeploymentParams;
+    string forkUrl;  
+    string emptyString;
+
+    address public protocolCouncilMultisig;
+    TimelockController public protocolTimelockController;
+    TimelockController public protocolTimelockController_BEIGEN;
+
+    address public beigenExecutorMultisig;
+
+    function run(string memory networkName) public virtual {
+        if (keccak256(abi.encodePacked(networkName)) == keccak256(abi.encodePacked("preprod-holesky"))) {
+            deployedContractsConfig = "script/configs/holesky/eigenlayer_addresses_preprod.config.json";
+            intialDeploymentParams = "script/configs/holesky/eigenlayer_preprod.config.json";
+            forkUrl = vm.envString("RPC_HOLESKY");
+            uint256 forkId = vm.createFork(forkUrl);
+            vm.selectFork(forkId);
+        } else if (keccak256(abi.encodePacked(networkName)) == keccak256(abi.encodePacked("testnet-holesky"))) {
+            deployedContractsConfig = "script/configs/holesky/eigenlayer_addresses_testnet.config.json";
+            intialDeploymentParams = "script/configs/holesky/eigenlayer_testnet.config.json";
+            forkUrl = vm.envString("RPC_HOLESKY");
+            uint256 forkId = vm.createFork(forkUrl);
+            vm.selectFork(forkId);
+        } else if (keccak256(abi.encodePacked(networkName)) == keccak256(abi.encodePacked("mainnet"))) {
+            deployedContractsConfig = "script/configs/mainnet/mainnet-addresses.config.json";
+            intialDeploymentParams = "script/configs/mainnet/mainnet-config.config.json"; 
+            forkUrl = vm.envString("RPC_MAINNET");
+            uint256 forkId = vm.createFork(forkUrl);
+            vm.selectFork(forkId);
+        }
+
+        if (keccak256(abi.encodePacked(networkName)) == keccak256(abi.encodePacked("local"))) {
```

Comment:
`else if`

Reply:
done in https://github.com/Layr-Labs/eigenlayer-contracts/pull/794/commits/d5844681cdfcc2a90b80a80ac97d440e1bf39736